src/testing/README.md

```diff
@@ -0,0 +1,60 @@
+## VOPR Output
+
+### Columns
+
+1. Replica index.
+2. Event:
+    - `$`: crash
+    - `^`: recover
+    - ` `: commit
+    - `[`: checkpoint start
+    - `]`: checkpoint done
+3. Role (according to the replica itself):
+    - `/`: primary
+    - `\`: backup
+    - `|`: standby
+    - `#`: (crashed)
+4. Status:
+    - The column (e.g. `.   ` vs `   .`) corresponds to the replica index. (This can help identify events' replicas at a quick glance.)
+    - The symbol indicates the `replica.status`.
+    - `.`: `normal`
+    - `v`: `view_change`
+    - `r`: `recovering`
+    - `h`: `recovering_head`
+5. View: e.g. `74V` indicates `replica.view=74`.
+6. Commit: e.g. `150/160C` indicates `replica.commit_min=150` and `replica.commit_max=160`.
+7. Journal op: e.g. `87:150Jo` indicates that the minimum op in the journal is `87` and the maximum is `150`.
+8. Journal dirty: `0Jd` indicates that the journal has 0 dirty headers.
+9. Journal faulty: `0Jf` indicates that the journal has 0 faulty headers.
+10. WAL prepare ops: e.g. `85:149Wo` indicates that the op of the oldest prepare in the WAL is `85` and the op of the newest prepare in the WAL is `149`.
+11. Grid blocks free: e.g. `122695Gf` indicates that the grid has `122695` blocks free.
+12. Pipeline prepares (primary-only): e.g. `2/4Pp` indicates that the primary's pipeline has 2 prepares queued, out of a capacity of 4.
+13. Pipeline requests (primary-only): e.g. `0/3Pq` indicates that the primary's pipeline has 0 requests queued, out of a capacity of 3.
```

Comment:
Do we need capacities here? they are constant, right?

Reply:
They are comptime-constant, but ideally we will eventually fuzz different capacities for these (the prepare queue and request queue).

e.g. the request queue is entirely optional — I sometimes manually fuzz it at zero.
And increasing/decreasing the prepare queue size can make different behaviors more/less likely.
In particular, there is different behavior when `prepare queue < batch multiple` vs `prepare queue > batch multiple`.